src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java

```diff
@@ -105,11 +115,20 @@ public static void waitForPipelineToTransitionToStatus(String pipelineStatus, lo
       CdfPipelineRunLocators.locatePipelineStatus(pipelineStatus), timeoutInSeconds);
   }
 
-  /**
+    /**
    * Wait till the Pipeline's status changes (from Running) to either Succeeded, Failed or Stopped within the
    * Timeout: {@link ConstantsUtil#PIPELINE_RUN_TIMEOUT_SECONDS}
    */
-  public static void waitTillPipelineRunCompletes() {
+  public static void waitTillPipelineRunCompletes() throws InterruptedException, IOException {
+    // Adding a page refresh in case tests are running on CDF to update the pipeline status.
+    if (Boolean.parseBoolean(SeleniumHelper.readParameters(ConstantsUtil.TESTONCDF)) ||
+      Boolean.parseBoolean(SeleniumHelper.readParameters(ConstantsUtil.TESTONHDF))) {
+      do {
+        PageHelper.refreshCurrentPage();
+        Thread.sleep(1000 * 60);
```

Comment:
Have we tried using any dynamic waits here?

Reply:
Didn't tried before, used now. (https://github.com/cloudsufi/cdap-e2e-tests/pull/2/commits/2fa8b80cc365818abeaf5bec5c89ab1185aa83dc#diff-cfa45894e4592a5d6ed32f485ed2594ae01b9a3b154e60820a599164e0b52e80R128)